pkg/config/mesh/mesh.go

```diff
@@ -40,63 +40,144 @@ var (
+		BinaryPath:            constants.BinaryPathFilename,
+		StatsdUdpAddress:      "",
+		EnvoyMetricsService:   &meshconfig.RemoteService{Address: ""},
+		EnvoyAccessLogService: &meshconfig.RemoteService{Address: ""},
+		CustomConfigFile:      "",
+		StatNameLength:        189,
+		StatusPort:            15020,
 	}
 }
 
+var (
+	TestMode = false
+)
+
 // DefaultMeshConfig configuration
 func DefaultMeshConfig() meshconfig.MeshConfig {
 	proxyConfig := DefaultProxyConfig()
+	if TestMode {
+		return meshconfig.MeshConfig{
+			IngressClass:                      "istio",
+			IngressControllerMode:             meshconfig.MeshConfig_STRICT,
+			ReportBatchMaxTime:                types.DurationProto(1 * time.Second),
+			ReportBatchMaxEntries:             100,
+			MixerCheckServer:                  "",
+			MixerReportServer:                 "",
+			DisablePolicyChecks:               true,
+			PolicyCheckFailOpen:               false,
+			SidecarToTelemetrySessionAffinity: false,
+			RootNamespace:                     constants.IstioSystemNamespace,
+			ProxyListenPort:                   15001,
+			ConnectTimeout:                    types.DurationProto(10 * time.Second),
+			IngressService:                    "istio-ingressgateway",
+			EnableTracing:                     true,
+			AccessLogFile:                     "/dev/stdout",
+			AccessLogEncoding:                 meshconfig.MeshConfig_TEXT,
+			DefaultConfig:                     &proxyConfig,
+			SdsUdsPath:                        "",
+			EnableSdsTokenMount:               false,
+			TrustDomain:                       "cluster.local",
+			TrustDomainAliases:                []string{},
+			Certificates:                      []*meshconfig.Certificate{},
+			DefaultServiceExportTo:            []string{"*"},
+			DefaultVirtualServiceExportTo:     []string{"*"},
+			DefaultDestinationRuleExportTo:    []string{"*"},
+			OutboundTrafficPolicy:             &meshconfig.MeshConfig_OutboundTrafficPolicy{Mode: meshconfig.MeshConfig_OutboundTrafficPolicy_ALLOW_ANY},
+			DnsRefreshRate:                    types.DurationProto(5 * time.Second), // 5 seconds is the default refresh rate used in Envoy
+			ProtocolDetectionTimeout:          types.DurationProto(100 * time.Millisecond),
+			EnableAutoMtls:                    &types.BoolValue{Value: false},
+			ThriftConfig:                      &meshconfig.MeshConfig_ThriftConfig{},
+			LocalityLbSetting:                 &v1alpha3.LocalityLoadBalancerSetting{},
+			ClusterLocalNamespaces:            append(make([]string, 0), defaultClusterLocalNamespaces...),
+		}
+	}
+	// Defaults matching the standard install
+	// order matches the generated mesh config.
 	return meshconfig.MeshConfig{
-		IngressClass:                      "istio",
-		IngressControllerMode:             meshconfig.MeshConfig_STRICT,
-		ReportBatchMaxTime:                types.DurationProto(1 * time.Second),
-		ReportBatchMaxEntries:             100,
+		EnableTracing:               true,
```

Comment:
Could you describe what did you change, you know there are too many fields and hard to review one by one.

Reply:
I changed the order to match what we have in mesh config, and changed the values to match
what we set in the default install. 

You can look side-by-side at a mesh config map using the default install - not sure yet how 
to keep them in sync as changes are made to install, but the goal is to have an empty config map by default, to allow user to make changes without installer.